src/main/java/net/buildtheearth/buildteam/components/generator/field/FieldScripts.java

```diff
@@ -0,0 +1,499 @@
+                    currentLowest = yellowWoolBlock.getLocation().getBlockX();
+                    westernMost = yellowWoolBlock;
+                }
+
+                if (yellowWoolBlock.getLocation().getBlockX() > currentHighest) {
+                    currentHighest = yellowWoolBlock.getLocation().getBlockX();
+                    easternMost = yellowWoolBlock;
+                }
+            }
+
+            // Check to make sure 2 yellow wool blocks were correctly found
+            if (westernMost == null || easternMost == null) {
+                p.sendMessage("§cSomething went wrong while processing line data!");
+                p.sendMessage(" ");
+                p.sendMessage("§cPlease contact one of the developers!");
+                return;
+            }
+
+
+            // Get the target length of the new line
+            double targetLength = 1;
+
+            double diagonal1 = region.getMaximumPoint().subtract(region.getMinimumPoint()).length();
+            Vector boundingBoxPoint3 = new Vector(region.getMinimumPoint().getX(), region.getMaximumPoint().getY(), region.getMaximumPoint().getZ());
+            Vector boundingBoxPoint4 = new Vector(region.getMaximumPoint().getX(), region.getMinimumPoint().getY(), region.getMinimumPoint().getZ());
+            double diagonal2 = boundingBoxPoint4.subtract(boundingBoxPoint3).length();
+
+            targetLength = Math.max(diagonal1, diagonal2);
+
+
+            // Get two new points of the extended line
+            Vector point1 = new Vector(westernMost.getX(), westernMost.getY(), westernMost.getZ());
+            Vector point2 = new Vector(easternMost.getX(), easternMost.getY(), easternMost.getZ());
+
+            Vector[] extendedVectors = interpolateVectors(point1, point2, targetLength);
+            Vector extendedPoint1 = extendedVectors[0];
+            Vector extendedPoint2 = extendedVectors[1];
+
+
+            // Draw the first line
+            commands.add("//sel cuboid");
+            commands.add("//gmask !air");
+
+            commands.add("//pos1 " + extendedPoint1.getBlockX() + "," + (extendedPoint1.getBlockY() - 1) + "," + extendedPoint1.getBlockZ());
+            commands.add("//pos2 " + extendedPoint2.getBlockX() + "," + (extendedPoint2.getBlockY() - 1) + "," + extendedPoint2.getBlockZ());
+            commands.add("//line 35:4");
+            operations++;
+
+            commands.add("//expand 10 up");
+            commands.add("//expand 10 down");
+
+            // Make sure the line completely covers the required surface
+            commands.add("//gmask !7,0");
+            for (int i = maxHeight; i < maxHeight + 5; i++) {
+                commands.add("//replace >35:4 35:4");
+                commands.add("//replace <35:4 35:4");
+                operations++;
+            }
+
+            // Reselect original region
+            Generator.createCuboidSelection(commands, region.getMaximumPoint(), region.getMinimumPoint());
```

Comment:
Why do you create the field in the Cuboid Selection and not the Poly Selection? 
If you would create the field in the Poly selection you would not need the bedrock remembering blocks afterwards to restore the fields original shape.
I think doing it your way will lead to all sorts of bugs since you are working in an area that should not be touched by the generator. 

What if the player has other stuff or a different field right next to his poly selection? This could get destroyed right now.

Reply:
This is because if the line isn't the widest part of the polygon, extending it won't work correctly. What I could maybe do is get all the blocks in the polygon selection, all the blocks in the cuboid selection, then all blocks that aren't in the poly selection get reset to their original state? This would probably eat performance though especially for large selections.

Reply:
Thats right. What if you do the pattern in the air like +10 blocks above the surface?
1. check if any block is not air in distance=10 blocks above the surface (match the elevation on the ground)
2. if there is a block that is not air, add distance += 1 and try again
3. draw the pattern of the field with a different color
4. draw the pattern on the ground but only in the polygon shape
5. remove the field pattern in the air